editor/modules/runner/game-input.ts

```diff
@@ -0,0 +1,140 @@
+import {Injectable} from '@angular/core';
+import {KeyboardLayoutDetector} from './keyboard-layout';
+import {Keys} from './keyboard-layout';
+import {MouseHandler, KeyHandler} from '../../components';
+import {LycanService} from './lycan.service';
+import {Direction} from './lycan.service';
+import {Player} from './player';
+
+
+export const enum Actions {
+    UP = 1,
+    DOWN = 2,
+    LEFT = 3,
+    RIGHT = 4
+}
+
+@Injectable()
+export class GameInput implements MouseHandler, KeyHandler {
+
+    private keyboard = new KeyboardLayoutDetector();
+    private actions_active: Actions[] = [];
+
+    constructor(private lycan: LycanService, private player: Player) {
+        this.keyboard.register('qwerty', Actions.UP, Keys.W);
+        this.keyboard.register('qwerty', Actions.DOWN, Keys.S);
+        this.keyboard.register('qwerty', Actions.LEFT, Keys.A);
+        this.keyboard.register('qwerty', Actions.RIGHT, Keys.D);
+        this.keyboard.register('azerty', Actions.UP, Keys.Z);
+        this.keyboard.register('azerty', Actions.DOWN, Keys.S);
+        this.keyboard.register('azerty', Actions.LEFT, Keys.Q);
+        this.keyboard.register('azerty', Actions.RIGHT, Keys.D);
```

Comment:
Can we rely on some library for that? There are a lot more keyboard layout in the wild (for instance, bépo and dvorak d:)